### PR TITLE
Change default value of `namespace` in `task.kubernetes` to be None

### LINKED ISSUE #42957: Kubernetes task decorator default namespace behaves different than KubernetesPodOperator when in cluster
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.9.2

### What happened?

Given these settings for KuberneteseExecutor:

Section | Key | Value
-- | -- | --
kubernetes_executor | in_cluster | True |
kubernetes_executor | namespace | redacted_not_default |

the behavior of `@task.kubernetes()` vs `KubernetesOperator()` when not setting namespace differs:

- `@task.kubernetes()` defaults to `namespace="default"`
- `KuberntesPodOperator()` defaults to `namespace=None` which uses the cluster namespace when `in_cluster` is true.

### What you think should happen instead?

`KubernetesPodOperator` and `@task.kubernetes` should use the in-cluster namespace by default (i.e., the worker pod's namespace).

### How to reproduce

1. Use a namespace that isn't `default`, let's call it `foobar` for this example
2. Set the service account to only allow pod API access to `foobar` namespace.
3. Run this DAG

    ```python
    with DAG(...):
        @task.kubernetes(image="hello-world"):
        def foo():
            ...
        taskA = foo()
    
        taskB = KubernetesPodOperator(image="hello-world")
    ```

You'll see K8s API error for the wrong namespace. In my case, the KPO decorator is trying to access the `airflow` namespace despite the pod is running in the correct namespace

```
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods is forbidden: User \"system:serviceaccount:airflow:airflow\" cannot list resource \"pods\" in API group \"\" in the namespace \"airflow\"","reason":"Forbidden","details":{"kind":"pods"},"code":403}
```

Here's the KPO decorator task's pod spec:
```
metadata:
  annotations:
    dag_id: lensing_model_classification
    run_id: manual__2024-10-11T22:11:31.074729+00:00
    task_id: redacted
    try_number: '4'
  labels:
    airflow-worker: '0'
    airflow_version: 2.9.2
    dag_id: redacted
    kubernetes_executor: 'True'
    run_id: manual__2024-10-11T221131.0747290000-e1a8551c6
    task_id: redacted
    try_number: '4'
  name: redacted
  namespace: 'foobar'
...
```

### Operating System

Amazon Linux 2

### Versions of Apache Airflow Providers

apache-airflow-providers-cncf-kubernetes==7.14.0

### Deployment

Other 3rd-party Helm chart

### Deployment details

_No response_

### Anything else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by romsharon98**
closes: https://github.com/apache/airflow/issues/42957

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: providers/src/airflow/providers/cncf/kubernetes/decorators/kubernetes.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ class _KubernetesDecoratedOperator(DecoratedOperator, KubernetesPodOperator):
     # there are some cases we can't deepcopy the objects (e.g protobuf).
     shallow_copy_attrs: Sequence[str] = ("python_callable",)
 
-    def __init__(self, namespace: str = "default", use_dill: bool = False, **kwargs) -> None:
+    def __init__(self, namespace: str | None = None, use_dill: bool = False, **kwargs) -> None:
```
Comment (eladkal) [inline providers/src/airflow/providers/cncf/kubernetes/decorators/kubernetes.py:68]: do we consider this breaking change or a bug fix?

Comment (amoghrajesh) [reply providers/src/airflow/providers/cncf/kubernetes/decorators/kubernetes.py:68]: Looks like a breaking change to me, it will affect users who do not initialise the decorator namespace right?

Comment (romsharon98) [reply providers/src/airflow/providers/cncf/kubernetes/decorators/kubernetes.py:68]: Added a warning in changelog and bump the provider version.

Comment (amoghrajesh) [reply providers/src/airflow/providers/cncf/kubernetes/decorators/kubernetes.py:68]: Thank you

File: providers/src/airflow/providers/cncf/kubernetes/CHANGELOG.rst [hunk 2]
```diff
@@ -27,6 +27,13 @@
 Changelog
 ---------
```
Comment (amoghrajesh) [inline providers/src/airflow/providers/cncf/kubernetes/CHANGELOG.rst:29]: Is this format correct?
CC @eladkal

File: providers/src/airflow/providers/cncf/kubernetes/CHANGELOG.rst [hunk 3]
```diff
@@ -27,6 +27,13 @@
...
 
+main
+.....
+
+.. warning::
+  Set the default value of ``namespace`` in _KubernetesDecoratedOperator to ``None``, so it uses the cluster namespace when ``in_cluster`` is True. To retain the previous behavior, set ``namespace="default"``
```
Comment (eladkal) [inline providers/src/airflow/providers/cncf/kubernetes/CHANGELOG.rst]: Are users aware of this private class? Can we maybe rephrase this to how users use the functionality?

Comment (amoghrajesh) [reply providers/src/airflow/providers/cncf/kubernetes/CHANGELOG.rst]: Right. They shouldnt care about it since it is used by the task operator. Maybe we can add / phrase it better to ask the users to provide the namespace in the operator.

**Review by eladkal — APPROVED** [review]

**Review by amoghrajesh — APPROVED** [review]
